lib/Basket.php

```diff
@@ -244,13 +244,15 @@ public function setAddressStatus( $isReady, $countryCode, $addressString )
      */
     public function getEUVatCountries()
     {
-        $ret = [ ];
-        foreach ($this->state->vatRates[ 'rates' ] as $cc => $info) {
+	$ret = [ ];
+        if(!empty($this->state->vatRates[ 'rates' ])){
```

Comment:
$this->state->vatRates being empty would be a bug so we shouldn't be working around it here

Reply:
So this is addressing the following warnings on going to the dev server with a checked out patternseek:

Jun 27 13:22:35 centralindex wolf-app[224389]: PHP Warning:  Undefined array key "rates" in /home/webuser/patternseek/ecommerce/lib/Basket.php on line 248
Jun 27 13:22:35 centralindex wolf-app[224389]: default.WARNING: E_WARNING: foreach() argument must be of type array|object, null given {"code":2,"message":"foreach() argument must be of type array|object, null given","file":"/home/webuser/patternseek/ecommerce/lib/Basket.php","line":248} []

Reply:
So this is triggered by an error dev mode whereby the vat rates are not coming down.
http://bugz.getinstance.com/show_bug.cgi?id=5914
When this bug is fixed I'll checkin a change that removes this fix.. so change here pending..

Reply:
change checked in below